Homework 3/src/SortedLinkedList.java

```diff
@@ -0,0 +1,190 @@
+/**
+ * Implements a Linked List data structure that is always sorted and uses only recursion.
+ * Andrew Id:
+ *
+ * @author
+ */
+public class SortedLinkedList implements MyListInterface {
+
+    private class Node {
+        String data;
+        Node next;
+
+        private Node() {
+            data = null;
+            next = null;
+        }
+    }
+
+    private Node head;
+
+    public SortedLinkedList(String[] array) {
+        if (array != null && array.length > 0)
+            constructorHelper(array, 0);
+    }
+
```

Comment:
Is this method used anywhere else? If not can It not be combined in the constructor itself?

You could use an early return to make the code cleaner in the constructor.

Reply:
No loops are allowed, iterations must be recursive.